test/authorizers/RequesterAuthorizerWithAirnode.sol.js

```diff
@@ -292,6 +304,122 @@ describe('RequesterAuthorizerWithAirnode', function () {
         });
       });
     });
+    // Let us demonstrate meta-calls as a proof of concept
+    context('Sender using a meta-call signed by the Airnode address', function () {
+      context('Timestamp extends whitelist expiration', function () {
+        it('extends whitelist expiration', async function () {
+          await accessControlRegistry
+            .connect(airnodeWallet)
+            .renounceRole(whitelistExpirationExtenderRole, airnodeAddress, { gasLimit: 1000000 });
+          let whitelistStatus;
+          whitelistStatus = await requesterAuthorizerWithAirnode.airnodeToEndpointIdToRequesterToWhitelistStatus(
+            airnodeAddress,
+            endpointId,
+            roles.requester.address
+          );
+          expect(whitelistStatus.expirationTimestamp).to.equal(0);
+          expect(whitelistStatus.indefiniteWhitelistCount).to.equal(0);
+          const expirationTimestamp = 1000;
+
+          const from = airnodeAddress;
+          const to = requesterAuthorizerWithAirnode.address;
+          const data = requesterAuthorizerWithAirnode.interface.encodeFunctionData('extendWhitelistExpiration', [
+            airnodeAddress,
+            endpointId,
+            roles.requester.address,
+            expirationTimestamp,
+          ]);
+          const metaCallExpirationTimestamp = (await testUtils.getCurrentTimestamp(hre.ethers.provider)) + 3600;
+
+          const domainName = 'ExpiringMetaCallForwarder';
+          const domainVersion = '1.0.0';
+          const domainChainId = (await hre.ethers.provider.getNetwork()).chainId;
+          const domainAddress = expiringMetaCallForwarder.address;
+
+          const domain = {
+            name: domainName,
+            version: domainVersion,
+            chainId: domainChainId,
+            verifyingContract: domainAddress,
+          };
+          const types = {
+            ExpiringMetaCall: [
+              { name: 'from', type: 'address' },
+              { name: 'to', type: 'address' },
+              { name: 'data', type: 'bytes' },
+              { name: 'expirationTimestamp', type: 'uint256' },
+            ],
+          };
+          const value = {
+            from,
+            to,
+            data,
+            expirationTimestamp: metaCallExpirationTimestamp,
+          };
+          const signature = await airnodeWallet._signTypedData(domain, types, value);
+
+          await expect(expiringMetaCallForwarder.connect(roles.randomPerson).execute(value, signature))
+            .to.emit(requesterAuthorizerWithAirnode, 'ExtendedWhitelistExpiration')
+            .withArgs(airnodeAddress, endpointId, roles.requester.address, airnodeAddress, expirationTimestamp);
```

Comment:
🤯 this is so cool!

The trick here is that _msgSender() == from == airnodeAddress == [account](https://github.com/api3dao/airnode-protocol-v1/blob/meta-tx/contracts/whitelist/WhitelistRolesWithAirnode.sol#L97)

Reply:
Yeah it was surprisingly easy to add. As a note, the `OrIsAirnode` part of `hasWhitelistExpirationExtenderRoleOrIsAirnode` is why I figured we don't need to add meta-tx support to AccessControlRegistry. Not sure if this is entirely valid though, an Airnode operator may still want to create a role and assign it to a contract, and not want to send any transactions themselves.